iroha/src/dex.rs

```diff
@@ -1029,7 +1004,7 @@ pub mod isi {
                     (amount_a_desired, amount_b_optimal)
                 } else {
                     let amount_a_optimal = quote(amount_b_desired.clone(), reserve_b, reserve_a)?;
-                    assert!(amount_a_optimal <= amount_a_desired); // TODO: consider not using assert
+                    assert!(amount_a_optimal <= amount_a_desired);
```

Comment:
Is this `assert` used in a test? If not, please replace with `debug_assert!`, otherwise this may cause panic

Reply:
Good point, should've used some kind of ensure macro for such checks.